CASMTRIAGE-7763: Workaround for cray-console-node in CLBO state (#5711)
Cray-HPE/docs-csm: troubleshooting/known_issues/cray_console_node_in_CLBO_state.md

Reviewer: The name of this file should match the title of the page. We have some pages out there that do not follow this rule, but generally they are older ones which were created before we started following that convention. If you take my title change suggestion below, then the file should be renamed to `cray-console-node_pods_in_CrashLoopBackOff.md`

Reviewer: Without any links to this page, it will be very hard to find. At a minimum it should be added to the `troubleshooting/README.md` file. The pages under `operations/conman/` should also be reviewed, to see if there are any locations in there where it makes sense to link it.

> # `CRAY-CONSOLE-NODE` pods intermittently goes into CLBO state after fresh CSM install

Reviewer: The title of the page should be relatively short. 
```suggestion
# `cray-console-node` pods in `CrashLoopBackOff`
```

> ## Issue Description

Reviewer: In our docs, we've been following the HPE tech writing style guide for how to capitalize section headers.
The main header of the page should use title casing (that is, capitalize most nouns). All subheaders should follow sentence casing (as in, the way you would capitalize a sentence -- capitalize only the first word and any proper nouns). 
```suggestion
## Issue description
```

> After installing CSM 1.6, `CRAY-CONSOLE-NODE` pods intermittently goes into CLBO state. This happens because of 

Reviewer: Is this actually intermittent? As in, will it recover on its own? Or is it something that will keep happening once you encounter it?

> permission issue with /var/log/conman directory inside the pods/container.

Reviewer: ```suggestion
After installing CSM 1.6, `cray-console-node` Kubernetes pods may intermittently go into a `CrashLoopBackOff` state.
This happens because of a permission issue with the `/var/log/conman` directory inside the container.
```

> ## Error Identification

Reviewer: ```suggestion
## Error identification
```

    
When the issue occurs, the `CRAY-CONSOLE-NODE` pods will be in CLBO state. The logs from the pod will be throwing errors as follows:

```text
2023/02/03 16:30:37 Starting a new instance of command
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s7b0n0
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s21b3n0
2023/02/03 16:30:37 Starting command process
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s9b0n0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x701979]

goroutine 57 [running]:
main.watchConsoleLogFile({0x835878, 0xc0003185c0}, {0xc000040920, 0xd})
    /usr/local/golib/src/console_node/logAggregation.go:154 +0x2b9
created by main.aggregateFile
    /usr/local/golib/src/console_node/logAggregation.go:68 +0x14b
```

## Fix Description

The workaround is to assign the correct permissions to the /var/log/conman directory inside the pods/container.

> 1. Find the cray-console-node pod IDs.

Reviewer 2: These all need to follow the convention to say where these commands are being run: (`ncn-mw#`) in this case.
This needs to be done for every step with a command in the doc.

Author: Done

    ```bash
    kubectl get pods -n services --no-headers -o wide | grep cray-console-node | awk '{print $1}'
    ```
   Example output:
    ```text
    cray-console-node-0
    cray-console-node-1
    ```

> 1. Log into the cray-console-node pods using their IDs.

Reviewer 2: This is a shared PVC - it only needs to be done on one pod, not all of them. Also could be done from the 'console-operator' as that also accessed the same PVC.

Author: Done

    ```bash

>     kubectl exec -n services -it CRAY-CONSOLE-NODE_POD_ID -- /bin/sh

Reviewer 2: Be consistent with '-' or '_' in the variable names.

Author: Done

    ```
1. Change the permission of the /var/log/ directory.

    ```bash
    chmod -R 700 /var/log/
    ```
1. Verify the permission of the /var/log/ directory.

    ```bash
    ls -ld /var/log/
    ```
    Example output:
    ```text
    drwx------ 2 nobody nobody 15 Jan 30 22:05 conman
    drwx------ 2 nobody nobody 21 Jan 30 22:05 conman.old
    drwx------ 2 nobody nobody  1 Dec  1  2023 console
    ```
1. Exit the pod.

    ```bash
    exit
    ```

> 1. Run the following command and notice that cray-console-pods are in running state.

Reviewer 2: 'Wait for the `cray-console-node-N` pods to return to a 'Running' state.'

Author: Done

# `CRAY-CONSOLE-NODE` pods intermittently goes into CLBO state after fresh CSM install

## Issue Description

> After installing CSM 1.6, `CRAY-CONSOLE-NODE` pods intermittently goes into CLBO state. This happens because of

Reviewer: Is this problem actually intermittent? As in, does it recover on its own once it happens? Or is it something that once it happens, it will keep happening until the workaround is applied?

> After installing CSM 1.6, `CRAY-CONSOLE-NODE` pods intermittently goes into CLBO state. This happens because of
> permission issue with /var/log/conman directory inside the pods/container.

Reviewer: ```suggestion
After installing CSM 1.6, `cray-console-node` Kubernetes pods intermittently go into a `CrashLoopBackOff` state.
This happens because of a permission issue with the `/var/log/conman` directory inside the container.
```

## Error Identification

> When the issue occurs, the `CRAY-CONSOLE-NODE` pods will be in CLBO state. The logs from the pod will be throwing errors as follows:

Reviewer: ```suggestion
The symptom of this problem is that the `cray-console-node` pods will be in a `CrashLoopBackOff` state.
The pod logs will contain messages resembling the following:
```

```text
2023/02/03 16:30:37 Starting a new instance of command
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s7b0n0
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s21b3n0
2023/02/03 16:30:37 Starting command process
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s9b0n0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x701979]

goroutine 57 [running]:
main.watchConsoleLogFile({0x835878, 0xc0003185c0}, {0xc000040920, 0xd})
    /usr/local/golib/src/console_node/logAggregation.go:154 +0x2b9
created by main.aggregateFile
    /usr/local/golib/src/console_node/logAggregation.go:68 +0x14b
```

> ## Fix Description

Reviewer: ```suggestion
## Fix procedure
```

> The workaround is to assign the correct permissions to the /var/log/conman directory inside the pods/container.

Reviewer: ```suggestion
The workaround is to assign the correct permissions to the `/var/log/conman` directory inside the container. This only needs to be done for a single pod, since the directory is shared between them.
```

2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s7b0n0
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s21b3n0
2023/02/03 16:30:37 Starting command process
2023/02/03 16:30:37 Starting to parse file: /var/log/conman/console.x3000c0s9b0n0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x701979]

goroutine 57 [running]:
main.watchConsoleLogFile({0x835878, 0xc0003185c0}, {0xc000040920, 0xd})
    /usr/local/golib/src/console_node/logAggregation.go:154 +0x2b9
created by main.aggregateFile
    /usr/local/golib/src/console_node/logAggregation.go:68 +0x14b
```

## Fix Description

The workaround is to assign the correct permissions to the /var/log/conman directory inside the pods/container.

1. (`ncn-mw#`) Find the `cray-console-node` pod IDs.

    ```bash
    kubectl get pods -n services --no-headers -o wide | grep cray-console-node | awk '{print $1}'
    ```

   Example output:

    ```text
    cray-console-node-0
    cray-console-node-1
    ```

1. (`ncn-mw#`) Log into one of the `cray-console-node` pods using their IDs.

    ```bash

>     kubectl exec -n services -it CRAY-CONSOLE-NODE-POD-ID -- /bin/sh

Reviewer: Not a big deal -- the current procedure is fine, and I don't think this change is necessary. Just wanted to mention that you could simplify this procedure if desired and consolidate this step with the next 3:
```
kubectl exec -n services CRAY-CONSOLE-NODE-POD-ID -- /bin/sh -c 'chmod -Rv 700 /var/log/ && ls -ld /var/log/'
```

    ```

> 1. Change the permission of the /var/log/ directory.

Reviewer: ```suggestion
1. (`pod#`) Change the permission of the `/var/log/` directory.
```

    ```bash
    chmod -R 700 /var/log/
    ```

> 1. Verify the permission of the /var/log/ directory.

Reviewer: ```suggestion
1. (`pod#`) Verify the permission of the `/var/log/` directory.
```

    ```bash
    ls -ld /var/log/
    ```

    Example output:

    ```text
    drwx------ 2 nobody nobody 15 Jan 30 22:05 conman
    drwx------ 2 nobody nobody 21 Jan 30 22:05 conman.old

>     drwx------ 2 nobody nobody  1 Dec  1  2023 console

Reviewer: With the `ls` arguments you used, I think the output would be:
```
drwx------ 5 nobody nobody 3 Jan 20 13:20 /var/log/
```

I think if you want to get the output you are showing, you'd have to run
```
ls -ld /var/log/*
```

    ```

> 1. (`ncn-mw#`) Exit the pod.

Reviewer: ```suggestion
1. (`pod#`) Exit the pod.
```

    ```bash
    exit
    ```

> 1. (`ncn-mw#`) Run the following command and notice that `cray-console-node-N` pods are in running state.

Reviewer: ```suggestion
1. (`ncn-mw#`) Verify that the `cray-console-node` pods are now in the `Running` state.
```
